### ISSUE TITLE
PyQwt which installs into PyQt4 but fails because of cyclic dependencies

### ISSUE THREAD

**Issue by qknight** [NixOS/nixpkgs#4904]
i'm packaging PyQwt but all my attempts fail so far...
https://github.com/qknight/nixpkgs/commits/brickv-2.1.1

the last thing i came up with was to split PyQwt and PyQt4 installation directories but that fails, see:
<https://gist.github.com/qknight/56abc0fcdb59a017395c>

**i now assume that PyQwt must be installed into PyQt4 because i'm getting an error similar to the one described here (details see above gist)**:
<http://lists.cairographics.org/archives/cairo/2013-May/024334.html>

the issue now is that i can't simply use a symlink from PyQt4 to PyQwt because PyQwt depends on PyQt4 while buildtime. however, this creates a cyclic dependency and nix can't evaluate it.

**i need directions how to continue with PyQwt**

thanks in advance!

**Comment by iElectric** [comment]
@qknight any progress here?

**Comment by qknight** [comment]
this matter is currently on my 'low priority list'. but **the repositories of pyqwt have been moved to github** (from CVS to GIT) **with the current maintainers blessings**. this is only a first step and now we have to figure a way on how to solve this issue. there are two possible ways:

1. rewrite pyqwt in a way that it is NOT an extension of pyqt but a standalone extension 

2. rewrite pyqwt to make it a 'real' part of pyqt so that it is a 'optional' feature of pyqt. this would mean that we have to get pyqwt into pyqt.

what we have to solve is this:
**currently pyqwt's installation routine has a mutual dependency on pyqt as it stateflly wants to write to the pyqt folder which will fail on nixos and should be considered a highly critical practice in general!**

right now i'm not sure what the best way of doing things here is.

**Comment by qknight** [comment]
seems i don't have the muse to do this right now. 

@domenkozar what do you think, could we two sprint this issue?

**Comment by domenkozar** [comment]
I looked into it and I have no idea what's going on, I'm not that interested into spending more time..

**Comment by qknight** [comment]
yes, that is also true for me altough i would have loved to have brickv (tinkerforge brick-viewer) working on nixos.

**Comment by domenkozar** [comment]
Did you solve this?

**Comment by qknight** [comment]
no. don't have the resources currently.

**Comment by FRidh** [comment]
Because this isn't a Nixpkgs issue I'm closing this for now. Of course, if you have questions, feel free to ask.
